src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.cs

```diff
@@ -118,19 +127,17 @@ public async Task OnCompilationEventsGeneratedAsync(ImmutableArray<CompilationEv
 
         private void OnCompilationEventsGenerated_NoLock(ImmutableArray<CompilationEvent> compilationEvents)
         {
-            Debug.Assert(_lazyAnalyzerActionCountsMap != null);
```

Comment:
💭  Why not keep this assertion here at the beginning of the method?

Reply:
Lazy nullable field is no longer expected to be read except in the corresponding non-nullable property getter, which asserts non-null field value.